src/ext/framework/help_commands.rs

```diff
@@ -0,0 +1,183 @@
+pub use ext::framework::command::{Command, CommandType, CommandGroup};
+
+use ::utils::Colour;
+use std::collections::HashMap;
+use std::sync::Arc;
+use ::client::Context;
+use ::model::Message;
+
+fn error_embed(ctx: &Context, message: &Message, input: &str) {
+    let _ = ctx.send_message(message.channel_id, |m| {
+        m.embed(|e| {
+            e.colour(Colour::dark_red())
+                .description(input)
+        })
+    });
+}
+
+#[allow(dead_code)]
+pub fn with_embeds(ctx: &Context,
+                   message: &Message,
+                   groups: HashMap<String, Arc<CommandGroup>>,
+                   args: Vec<String>) {
+    if args.len() > 0 {
+        let name = args.join(" ");
+        for (_, group) in groups {
+            let mut found: Option<(&String, &Command)> = None;
+            if let Some(ref prefix) = group.prefix {
+                for (command_name, command) in &group.commands {
+                    if name == format!("{} {}", prefix, command_name) {
+                        found = Some((command_name, command));
+                    }
+                }
+            } else {
+                for (command_name, command) in &group.commands {
+                    if name == command_name[..] {
+                        found = Some((command_name, command));
+                    }
+                }
+            };
+            if let Some((command_name, command)) = found {
+                if !command.help_available {
+                    error_embed(&ctx, &message, "**Error**: No help available.");
```

Comment:
Since `ctx` and `message` are already borrowed, `&` can be omitted, because right now it'll just be dereferenced by rustc